apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java

```diff
@@ -411,14 +411,19 @@ private boolean containsAspect(List<AspectDefinition> aspects, AspectDefinition
 
     @Override
     public List<NodeAssociation> getSourceAssociations(eu.xenit.apix.data.NodeRef ref) {
-        return nodeService.getSourceAssocs(c.alfresco(ref), RegexQNamePattern.MATCH_ALL)
-                .stream()
-                .map(alfPeerAssoc ->
-                        new NodeAssociation(
-                                c.apix(alfPeerAssoc.getSourceRef()),
-                                ref,
-                                c.apix(alfPeerAssoc.getTypeQName())))
-                .collect(Collectors.toList());
+        // Versionstore does not support sourceAssocs. For version nodes, do not do call, add empty list to result
+        List<NodeAssociation> sourceAssocs = null;
+        if ("versionStore".equals(ref.getStoreRefProtocol())) {
+            sourceAssocs = nodeService.getSourceAssocs(c.alfresco(ref), RegexQNamePattern.MATCH_ALL)
+                    .stream()
+                    .map(alfPeerAssoc ->
+                            new NodeAssociation(
+                                    c.apix(alfPeerAssoc.getSourceRef()),
+                                    ref,
+                                    c.apix(alfPeerAssoc.getTypeQName())))
+                    .collect(Collectors.toList());
+        }
+        return sourceAssocs == null ? new ArrayList<>() : sourceAssocs;
```

Comment:
IMO control flow is more readable like this:
```suggestion
        List<NodeAssociation> sourceAssocs = null;
        if ("versionStore".equals(ref.getStoreRefProtocol())) {
            return new ArrayList<>();
        }
        
        return nodeService.getSourceAssocs(c.alfresco(ref), RegexQNamePattern.MATCH_ALL)
                    .stream()
                    .map(alfPeerAssoc ->
                            new NodeAssociation(
                                    c.apix(alfPeerAssoc.getSourceRef()),
                                    ref,
                                    c.apix(alfPeerAssoc.getTypeQName())))
                    .collect(Collectors.toList());
```

Reply:
The red line in my suggestion is a bit misleading, sorry. My suggestion is intended to replace the entire new code block.